x/migration/keeper/msg_server_claim_morse_supplier.go

```diff
@@ -251,10 +254,55 @@ func (k msgServer) ClaimMorseSupplier(
 			ClaimedBalance:       morseClaimableAccount.TotalTokens(),
 			ClaimedSupplierStake: cosmostypes.Coin{},
 			SessionEndHeight:     sessionEndHeight,
-			Supplier:             supplier,
+			Supplier:             unbondedSupplier,
 		}, nil
 	}
 
+	claimedSupplierStake := morseClaimableAccount.GetSupplierStake()
+	claimedUnstakedBalance := morseClaimableAccount.GetUnstakedBalance()
+
+	// TODO_IN_THIS_COMMIT: comment...
+	// ... supplier began unbonding on Morse and the unbonding period has since elapsed...
+	// ... unstatked balance was minted to the shannonSignerAddr account...
+	// ... stake was minted to the shannonOwnerAddr account...
+	// ... no Supplier was staked...
+	// - use a lookup table to get the est. block times per network...
+	// - use the shared params and the block time to calculate the unstake session end height...
+	// - emit an unbonding start event...
+	if !morseClaimableAccount.UnstakingCompletionTime.IsZero() {
+		durationUntilUnstake := time.Until(morseClaimableAccount.UnstakingCompletionTime)
+		if durationUntilUnstake.Seconds() <= 0 {
```

Comment:
Anyway you can consolidate these three lines into a semantic helper inside of morseClaimableAccount?

Reply:
Same as https://github.com/pokt-network/poktroll/pull/1338#discussion_r2095161799